src/persist/src/postgres.rs

```diff
@@ -294,16 +316,22 @@ impl Consensus for PostgresConsensus {
             // 1. Our shard will always have _some_ data mapped to it.
             // 2. All operations that modify the (seqno, data) can only increase
             //    the sequence number.
-            let current = self.head(deadline, key).await?;
+            let current = self.head_tx(&tx, key).await?;
             if current.map_or(true, |data| data.seqno < seqno) {
-                return Err(ExternalError::from(anyhow!(
+                Err(ExternalError::from(anyhow!(
```

Comment:
I don't think we want to get rid of the early return here? isn't it surprising to commit the txn if we're returning an error?

oh, it's because we want to ensure we linearize it. this definitely deserves a comment!

Reply:
yeah - i mentioned this elsewhere but my reading of the pg docs is that you shouldn't use the results of any reads within a txn unless you've committed that txn successfully

Reply:
added a comment